pkg/ext/stores/tokens/tokens.go

```diff
@@ -0,0 +1,1321 @@
+	user, err := t.auth.UserName(ctx)
+	if err != nil {
+		return nil, apierrors.NewInternalError(err)
+	}
+	if !userMatch(user, token) {
+		return nil, apierrors.NewBadRequest("unable to create token for other user")
+	}
+	return t.SystemStore.Create(ctx, GVR.GroupResource(), token, options)
+}
+
+func (t *SystemStore) Create(ctx context.Context, group schema.GroupResource, token *ext.Token, options *metav1.CreateOptions) (*ext.Token, error) {
+	// ensure existence of the namespace holding our secrets. run once per store.
+	if !t.initialized {
+		_, err := t.namespaceClient.Create(&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: TokenNamespace,
+			},
+		})
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			return nil, err
+		}
+		t.initialized = true
+	}
+
+	ensureNameOrGenerateName(token)
+	// we check the Name directly. because of the ensure... we know that
+	// GenerateName is not set. as it squashes the name in that case.
+	if token.Name != "" {
+		// reject creation of a token which already exists
+		currentSecret, err := t.secretClient.Get(TokenNamespace, token.Name, metav1.GetOptions{})
+		if err == nil && currentSecret != nil {
+			return nil, apierrors.NewAlreadyExists(group, token.Name)
+		}
+	}
+
+	// reject user-provided token value, or hash
+	if token.Status.TokenValue != "" {
+		return nil, apierrors.NewBadRequest("User provided token value is not permitted")
+	}
+	if token.Status.TokenHash != "" {
+		return nil, apierrors.NewBadRequest("User provided token hash is not permitted")
+	}
+
+	// Get derived User information for the token.
+	//
+	// NOTE: The creation process is not given `AuthProvider`, `User-` and `GroupPrincipals`.
+	// ..... This information has to be retrieved from somewhere else in the system.
+	// ..... This is in contrast to the Norman tokens who get this information either
+	// ..... as part of the Login process, or by copying the information out of the
+	// ..... base token the new one is derived from. None of that is possible here.
+	//
+	// The token store gets the necessary information (auth provider and
+	// principal id) from the extras of the user stored in the context.
+	//
+	// The `UserPrincipal` data is filled in part from the above, and in
+	// part from the associated `User`s fields.
+	//
+	// `ProviderInfo` is not supported. Norman tokens have it as legacy fallback to hold the
+	// `access_token` data managed by OIDC-based auth providers. The actual primary storage for
+	// this is actually a regular k8s Secret associated with the User.
+
+	user, err := t.userClient.Get(token.Spec.UserID, metav1.GetOptions{})
+	if err != nil {
+		return nil, apierrors.NewInternalError(fmt.Errorf("failed to retrieve user %s: %w",
+			token.Spec.UserID, err))
+	}
+	if user == nil {
+		return nil, apierrors.NewInternalError(fmt.Errorf("failed to get user %s", token.Spec.UserID))
+	}
+
+	// Reject operation if the user is disabled.
+	if user.Enabled == nil || !*user.Enabled {
```

Comment:
same as previous comment. Does `user.Enabled == nil` means the user is disabled?

Reply:
Fixed with commit 73501deeb6e905088f0eab7040de8c26f348d4b9 (rebase!)